File: CS/DxSample/PersistentObjects/Event.cs
using DevExpress.Xpo;
using System;

namespace DxSample.PersistentObjects {
    public class Event :XPObject {
        public Event(Session session) : base(session) { }

        string fSubject;
        public string Subject {
            get { return fSubject; }
            set { SetPropertyValue<string>("Subject", ref fSubject, value); }
        }

        DateTime fStartTime;
        public DateTime StartTime {
            get { return fStartTime; }
            set { SetPropertyValue<DateTime>("StartTime", ref fStartTime, value); }
        }

        DateTime fEndTime;
        public DateTime EndTime {
            get { return fEndTime; }
            set { SetPropertyValue<DateTime>("EndTime", ref fEndTime, value); }
        }
    }
}

File: VB/DxSample/PersistentObjects/Event.vb
Imports DevExpress.Xpo
Imports System

Namespace DxSample.PersistentObjects
    Public Class [Event]
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private fSubject As String
        Public Property Subject() As String
            Get
                Return fSubject
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Subject", fSubject, value)
            End Set
        End Property

        Private fStartTime As Date
        Public Property StartTime() As Date
            Get
                Return fStartTime
            End Get
            Set(ByVal value As Date)
                SetPropertyValue(Of Date)("StartTime", fStartTime, value)
            End Set
        End Property

        Private fEndTime As Date
        Public Property EndTime() As Date
            Get
                Return fEndTime
            End Get
            Set(ByVal value As Date)
                SetPropertyValue(Of Date)("EndTime", fEndTime, value)
            End Set
        End Property
    End Class
End Namespace
